ql/src/java/org/apache/hadoop/hive/ql/DriverContext.java

```diff
@@ -206,6 +208,9 @@ public ValidWriteIdList getCompactionWriteIds() {
 
   public void setCompactionWriteIds(ValidWriteIdList compactionWriteIds) {
     this.compactionWriteIds = compactionWriteIds;
+    if (compactionWriteIds != null) {
+      conf.set(ValidTxnWriteIdList.COMPACTOR_VALID_TABLES_WRITEIDS_KEY, compactionWriteIds.toString());
```

Comment:
if not being used may be move this out of this patch.

Reply:
We currently maintain the validWriteIds in DriverContext.